### PR TITLE
enable odf console for mcg

### PR DESCRIPTION
**Pull request by naveenpaul1**
enable odf dashboard console for mcg 
@rexagod

### CODE REVIEW

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -208,19 +224,46 @@ func (r *ManagedMCGReconciler) reconcilePhases() (reconcile.Result, error) {
...
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.operatorConsole, func() error {
+		r.operatorConsole.Spec.Plugins = append(r.operatorConsole.Spec.Plugins, odfConsoleName)
+		return nil
+	})
+	r.Log.Info("Updqted Console resources")
+	return err
+}
+
+func (r *ManagedMCGReconciler) reconcileConsolePlugin() error {
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:259]: Are we using this function anywhere?

**Review by rexagod — CHANGES_REQUESTED** [review]
@naveenpaul1 PTAL

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:259]: No will remove it

File: main.go [hunk 2]
```diff
@@ -72,6 +78,14 @@ func addAllSchemes(scheme *runtime.Scheme) {
 
 	utilruntime.Must(noobaa.AddToScheme(scheme))
 
+	utilruntime.Must(consolev1.AddToScheme(scheme))
+	utilruntime.Must(consolev1alpha1.AddToScheme(scheme))
```
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply main.go]: removed unwanted schemas

File: controllers/managedmcg_controller.go [hunk 3]
```diff
@@ -477,3 +534,41 @@ func (r *ManagedMCGReconciler) unrestrictedGet(obj client.Object) error {
...
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      odfConsoleDeployment.Name,
+		Namespace: odfConsoleDeployment.Namespace,
+	}, odfConsoleDeployment)
+	if err != nil {
+		return err
+	}
+
+	operatorConsole := operatorv1.Console{}
+	_ = operatorConsole
```
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go]: removed

File: controllers/managedmcg_controller.go [hunk 4]
```diff
@@ -111,6 +119,16 @@ func (r *ManagedMCGReconciler) initReconciler(req ctrl.Request) {
...
+//+kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=*
+//+kubebuilder:rbac:groups=operator.openshift.io,resources=consoles,verbs=*
+
+//+kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="apps",resources=deployments/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:131]: Just one more concern, do we need these?

**Review by rexagod — CHANGES_REQUESTED** [review]

File: controllers/managedmcg_controller.go (hunk 4 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go]: its removed

**Pull request merged**